carbon-design-system/carbon-addons-iot-react
# feat(table): add overflow menu to batch actions

Closes #3321 

**Summary**

- Adds an overflow menu to the table Batch actions bar. This follows the same dynamic as the overflow menu in the table toolbar. All items appearing in the overflow menu must be passed `isOverflow:true`. It also adds support for `hidden`, `disabled`, `isDelete`, `hasDivider`, etc. to match the other usages.

**Change List (commits, features, bugs, etc)**

- update proptypes for table batch actions
- add `hidden`, `disabled` to batch actions (those not in overflow menu)
- add `isOverflow` and display those items that aren't hidden in the overflow menu
- add tests to confirm
- update snapshots
- update with selection and batch action story

**Acceptance Test (how to verify the PR)**

- Go to the [with selection and batch actions story](https://deploy-preview-3338--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-selection-and-batch-actions)
- confirm the overflow menu exists
- confirm the actions are fired for clicking on those items (that aren't disabled)
- confirm there are only 3 items showing in the overflow menu (the 4th one is hidden)
- confirm the "Expunge" item is red when hovered
- confirm "Reassign" is disabled
- confirm there is a divider above both "Reassign" and "Expunge" (`hasDivider`)

**Regression Test (how to make sure this PR doesn't break old functionality)**

- ensure batch actions still work as expected



**Things to look for during review**
